testsuite/classlibrary/TestLinkClock.sc

```diff
@@ -202,4 +202,140 @@ TestLinkClock : UnitTest {
 
 		semaphore.wait
 	}
+
+	test_LinkClock_meter_change_beats_keep_continuity {
+		var clock1, clock2,
+		tempo = 12, beatDurThresh = 0.4 / tempo,
+		secList1 = List.new, secList2 = List.new,
+		cond = Condition({ secList1.size == 20 and: { secList2.size == 20 } });
+
+		clock1 = LinkClock(tempo);
+		{
+			2.5.wait;  // offset clock starts
+			clock2 = LinkClock(tempo);
+			1.5.wait;
+			{
+				20.do {
+					secList2.add(clock2.seconds);
+					1.0.wait;
+				};
+				cond.signal;
+			}.fork(clock2);
+			20.do {
+				secList1.add(clock1.seconds);
+				clock1.setMeterAtBeat(rrand(2, 4), clock1.beats);
+				1.0.wait;
+			};
+			cond.signal;
+		}.fork(clock1);
+		cond.wait;
+		this.ifAsserts(
+			clock1.numPeers == 1,
+			"no extra clocks in session",
+			{
+				this.assert(
+					(secList1 absdif: secList2).every(_ <= beatDurThresh),
+					"LinkClock:setMeterAtBeat should not disrupt beat duration\nclock1: %\nclock2: %\n"
+					.format(secList1, secList2)
+				)
+			},
+			{
+				this.assert(false, "Make sure no other LinkClocks are running while testing");
+			}
+		);
+		clock1.stop;
+		clock2.stop;
+	}
+
+	// this test uses "magic number" wait statements
+	// because it takes a little time for a new LinkClock to stabilize
+	// and we have no signal to know when it's ready
+	test_LinkClock_sync_meter_aligns_barlines {
+		var numTrials = 5,
+		refClock, testClock, list = List.new, resp,
+		cond = Condition.new, outerCond = Condition.new;
+		refClock = LinkClock(10);
+		1.0.wait;  // refClock's time is not stable yet
+		refClock.schedAbs(4, Routine {
+			if(refClock.numPeers > 0) {
+				this.assert(false, "Make sure no other LinkClocks are running while testing");
+			} {
+				refClock.beatsPerBar = 3;
+				// we must wait until the reference clock finishes its initial sync attempt
+				resp = SimpleController(refClock).put(\resynced, {
+					resp.remove;
+					cond.unhang;
+				});
+				refClock.enableMeterSync;
+				cond.hang;
+				numTrials.do { |i|
+					if(i > 0) {  // don't wait first time
+						rrand(0.2, 3.0).wait;
+					};
+					testClock = LinkClock.new;
+					// wait to pick up the peer
+					resp = SimpleController(testClock).put(\numPeers, {
+						resp.remove;
+						cond.unhang;
+					});
+					cond.hang;
+					testClock.enableMeterSync;
+					resp = SimpleController(testClock).put(\meter, {
+						resp.remove;
+						cond.unhang;
+					});
+					cond.hang;
+					testClock.getMeterSync.queryMeter { |replies|
+						list.add(replies.every { |reply|
+							(reply[\queriedAtBeat] - testClock.baseBarBeat) % testClock.beatsPerBar
+							absdif: reply[\beatInBar] < 0.05
+						});
+						cond.unhang;
+					};
+					cond.hang;
+					testClock.stop;
+				};
+				this.assertEquals(
+					list.count { |bool| bool },
+					list.size,
```

Comment:
it would be simpler to keep an integer variable that you increment for every successful trial, and then assert that its value is equal to `numTrials`.